algo/src/test/kotlin/com/hartwig/actin/algo/evaluation/tumor/HasTripleNegativeBreastCancerTest.kt

```diff
@@ -0,0 +1,196 @@
+package com.hartwig.actin.algo.evaluation.tumor
+
+import com.hartwig.actin.algo.doid.DoidConstants
+import com.hartwig.actin.algo.evaluation.EvaluationAssert.assertEvaluation
+import com.hartwig.actin.datamodel.algo.EvaluationResult
+import com.hartwig.actin.datamodel.clinical.IhcTest
+import com.hartwig.actin.doid.TestDoidModelFactory
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class HasTripleNegativeBreastCancerTest {
+
+    val doidModel = TestDoidModelFactory.createMinimalTestDoidModel()
+    val function = HasTripleNegativeBreastCancer(doidModel)
+
+    @Test
+    fun `Should evaluate to undetermined when no tumor doids configured`() {
+        val evaluation = function.evaluate(TumorTestFactory.withDoids(emptySet()))
+        assertEvaluation(EvaluationResult.UNDETERMINED, evaluation)
+        assertThat(evaluation.undeterminedMessagesStrings()).containsExactly("Undetermined if triple negative breast cancer (tumor doids missing)")
+    }
+
+    @Test
+    fun `Should fail if tumor type is not breast cancer`() {
+        assertEvaluation(EvaluationResult.FAIL, function.evaluate(TumorTestFactory.withDoids(setOf(DoidConstants.COLORECTAL_CANCER_DOID))))
+    }
+
+    @Test
+    fun `Should pass if tumor doid is triple negative breast cancer`() {
+        assertEvaluation(
+            EvaluationResult.PASS,
+            function.evaluate(
+                TumorTestFactory.withDoids(
+                    setOf(
+                        DoidConstants.BREAST_CANCER_DOID,
+                        DoidConstants.TRIPLE_NEGATIVE_BREAST_CANCER_DOID
+                    )
+                )
+            )
+        )
```

Comment:
It might be worth defining a `val` for some of these cases so they can be written more concisely, e.g.
```suggestion
        val patient = TumorTestFactory.withDoids(
            setOf(DoidConstants.BREAST_CANCER_DOID, DoidConstants.TRIPLE_NEGATIVE_BREAST_CANCER_DOID)
        )
        assertEvaluation(EvaluationResult.PASS, function.evaluate(patient))
```